scripts/docker/dockerfile_clang-10

```diff
@@ -1,7 +1,7 @@
 # This script can be run with the following command:
 # docker build --build-arg branch=<your branch here, defaults to develop> -t "axom/tpls:clang-10" - < dockerfile_clang-10
 
-FROM axom/compilers:ubuntu-20-clang-10_08-24-21_07h-18m
+FROM axom/compilers:ubuntu-20-clang-10_07-10-23_21h-54m
```

Comment:
This was as minimal a change as i could do since this PR has gone on long enough. This is the same docker container just with a newer CMake so we could build RAJA/Umpire.